gulp/tasks/deploy.js

```diff
@@ -44,28 +45,51 @@ gulp.task('deploy:version', function() {
     .pipe(gulp.dest('.'));
 });
 
-// Generates compiled CSS and JS files and puts them in the dist/ folder
+// Generates compiled CSS and JS files and sourcemaps and puts them in the dist/ folder
 gulp.task('deploy:dist', ['sass:foundation', 'javascript:foundation'], function() {
   var cssFilter = filter(['**/*.css'], { restore: true });
-  var sourcemapFilter = filter(['**/*.css.map'], { restore: true });
   var jsFilter  = filter(['**/*.js'], { restore: true });
+  var cssSourcemapFilter = filter(['**/*.css.map'], { restore: true });
+  var jsSourcemapFilter = filter(['**/*.js.map'], { restore: true });
 
-  console.log(CONFIG.DIST_FILES)
   return gulp.src(CONFIG.DIST_FILES)
     .pipe(plumber())
+
+    // --- Source maps ---
+    // * Copy sourcemaps to the dist folder
+    // This is done first to avoid collision with minified-sourcemaps.
+    .pipe(cssSourcemapFilter)
+      .pipe(gulp.dest('./dist/css'))
+      .pipe(cssSourcemapFilter.restore)
+    .pipe(jsSourcemapFilter)
+      .pipe(gulp.dest('./dist/js'))
+      .pipe(jsSourcemapFilter.restore)
+
+    // --- Source files ---
+    // * Copy source files to dist folder
+    // * Create minified files
+    // * Create minified-sourcemaps based on standard sourcemaps.
+    //   Sourcemaps are initialized before the ".min" renaming to be able retrieve
+    //   original sourcemaps (based on the source name), then renamed manually.
     .pipe(cssFilter)
       .pipe(gulp.dest('./dist/css'))
-      .pipe(cleancss({ compatibility: 'ie9' }))
+      .pipe(sourcemaps.init({ loadMaps: true }))
       .pipe(rename({ suffix: '.min' }))
+      .pipe(cleancss({ compatibility: 'ie9' }))
+      .pipe(sourcemaps.write('.', {
+        mapFile: function(path) { return path.replace('.css.map', '.min.css.map'); }
```

Comment:
This generates `.min.min.map`, should be probably `.min.map`.

Reply:
No because `Sourcemaps are initialized before the ".min" renaming to be able retrieve original sourcemaps`. So original sourcemaps `xxx.css.map` are found because files are still like `xxx.css`, and this generate `xxx.css.map` sourcemaps after the minification we rename to `xxx.min.css.map`.

Reply:
![image](https://user-images.githubusercontent.com/827205/37626257-89ebfcd0-2bcf-11e8-9218-421fe1907886.png)

![image](https://user-images.githubusercontent.com/827205/37626237-70683080-2bcf-11e8-963a-1182a1f1de2a.png)

Reply:
Should be

```js
    .pipe(cssFilter)
      .pipe(gulp.dest('./dist/css'))
      .pipe(sourcemaps.init({ loadMaps: true }))
      .pipe(rename({ suffix: '.min' }))
      .pipe(cleancss({ compatibility: 'ie9' }))
      .pipe(sourcemaps.write('.'))
      .pipe(gulp.dest('./dist/css'))
      .pipe(cssFilter.restore)

    .pipe(jsFilter)
      .pipe(gulp.dest('./dist/js'))
      .pipe(sourcemaps.init({ loadMaps: true }))
      .pipe(rename({ suffix: '.min' }))
      .pipe(uglify())
      .pipe(sourcemaps.write('.'))
      .pipe(gulp.dest('./dist/js'));
```

Reply:
Where there are written does only matter, the initialization collects the needed files. ;-)

Reply:
hmm... this worked on my side. I'll check that again. Thanks for the review ;)

Reply:
https://github.com/gulp-sourcemaps/gulp-sourcemaps/blob/master/src/init/index.js#L134
https://github.com/gulp-sourcemaps/gulp-sourcemaps/blob/master/src/write/index.internals.js#L106